extensions/standard-processors/processors/TailFile.cpp

```diff
@@ -66,7 +66,8 @@ core::Property TailFile::StateFile("State File", "Specifies the file that should
                                    " what data has been ingested so that upon restart NiFi can resume from where it left off",
                                    "TailFileState");
 core::Property TailFile::Delimiter("Input Delimiter", "Specifies the character that should be used for delimiting the data being tailed"
```

Comment:
If a delimiter isn't specified we should consider this an error if we're using a delimiter mode. If size is being used we can have that specified and make this property an error condition. we should separate those ideologies entirely.

Reply:
I agree, I just did not want to include that in this PR, because that would modify the behaviour of the processor in a way that could not be made backward-compatible with default values.
Currently empty Delimiter means size-based without a size limit, and a non-empty Delimiter means delimiter mode.
If we add a mode property, and make the default of that property the delimiter mode, we would be backward-compatible with the current delimiter mode, but break the non-delimiter mode.
If the default of that property is size-based mode, and the default size is 0 (unlimited), we keep compatibility with the non-delimiter mode, but break compatibility with the delimiter mode.
I think it is a necessary move to make, but I do not know how such a change (that would require configuration changes on update) can be made.